### PR TITLE
mon/LogMonitor: 'log last' command

### PR DESCRIPTION
**Pull request by liewegas**
- also clean up LogSummary class

### CODE REVIEW

**Comment by liewegas** [comment]
The use of LogSummary could stand to be rewritten, but this is simple and works (for up to 1024 recent entries).

File: src/common/LogEntry.h [hunk 1]
```diff
@@ -97,24 +105,22 @@ WRITE_CLASS_ENCODER_FEATURES(LogEntry)
 struct LogSummary {
   version_t version;
   list<LogEntry> tail;
+  ceph::unordered_set<LogEntryKey> keys;
```
Comment (tchaikov) [inline src/common/LogEntry.h]: if we want to be spacial efficient, probably we can just keep the hashed `LogEntryKey` around?

File: src/mon/MonCommands.h [hunk 2]
```diff
@@ -197,6 +197,9 @@ COMMAND_WITH_FLAG("scrub", "scrub the monitor stores", \
 COMMAND("fsid", "show cluster FSID/UUID", "mon", "r", "cli,rest")
 COMMAND("log name=logtext,type=CephString,n=N", \
 	"log supplied text to the monitor log", "mon", "rw", "cli,rest")
+COMMAND("log last name=num,type=CephInt,range=1,req=false", \
```
Comment (tchaikov) [inline src/mon/MonCommands.h]: can we have a test in, for example, `src/test/mon/misc.sh` ?

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-06_21:54:14-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**